shared/src/components/text_input.tsx

```diff
@@ -24,12 +24,14 @@ interface TextInputProps {
   type?: string;
   min?: string;
   max?: string;
+  step?: number;
   label?: string;
   name?: string;
   value?: string;
   placeholder?: string;
   hasError?: boolean;
   onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onBlur?: () => void;
```

Comment:
Perhaps you can follow the same pattern as we used for the `CheckboxInput` and make the props extend `React.InputHTMLAttributes<HTMLInputElement>`. Then, any props that aren't specifically defined for our custom use just get passed down to the <input> element.

https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/38aefa64465c6073844682fb1297b37d0cf33205/shared/src/components/checkbox_input.tsx#L22